### ISSUE TITLE
Martian Saucers never despawn upon 'flying away'.

### ISSUE THREAD

**Issue by tlworks** [NyxStudios/TShock#1196]
So I do think that this bug has to do with Terraria itself, but whenever someone spawns a Martian Saucer (By the core, ID 395) and goes to spawn by using /spawn the Martian Saucer(s) will fly away into the sky and outside of the Map, they never despawn and they have to be butchered with /butcher in order for them to despawn.

This issue has been there ever since 1.3, since it actually happened before 1.3.1 already. This has NOT been tested with the martian event itself, only spawned outside of the event.

Using latest Bamboo TShock 4.3.14.0 on Terraria 1.3.1 ~ Confirmed issue on older versions of 1.3 as well.

**Comment by mistzzt** [comment]
Did you manage to reproduce it on TerrariaApi-Server without TShock? 
But it is difficult to spawn a npc without TShock... faint.
Maybe you can write a plugin which helps you reproduce this situation.

**Comment by tlworks** [comment]
It's easy to reproduce on TShock as you simply have to spawn a martian saucer and 'warp' away so the npc stops targeting you. It'll simply fly off into the air (Look on the map by pressing M) and you'll even see the boss icons go up out of the map and if you /butcher they will be defeated.

Yet again I don't know if this happens during the actual event, because if not there is no way to reproduce it on vanilla.

**Comment by Kojirremer** [comment]
Though I haven't really tested it with a clean install, it happens to me, but with the Eater of Worlds.The boss' map icon will fly somewhere out of reach in the map, and if someone closes in, it will fly further away until it exits the map... and stays out of bounds. I had to make Autobroadcast use /butcher once per hour since I noticed the server would eventually hang if those boss icons were left unattended.
I though it could be a client-sided issue, but while the icon is nested somewhere, no additional EoWs could be spawned by anyone.

However, I repeat: I haven't seen this on a clean install, because it's somewhat random. I can't really reproduce it, so even if I try with a clean install, I wouldn't know what to do to trigger the issue.
